File: C/C_ForExample/22-Function/S30/demo2.c
#include  <stdio.h>

int add(int,int);
int sub(int,int);
int calc(int (*fp)(int,int), int, int);  //声明 calc函数：第一个参数函数指针 (注意类型)

int add(int num1, int num2)
{
  return num1 + num2;
}

int sub(int num1, int num2)
{
  return num1 - num2;
}

int calc(int (*fp)(int,int), int num1, int num2)
{
    return (*fp)(num1, num2);
}

int main()
{
  printf("3 + 5 = %d\n",calc(add, 3, 5));
  printf("3 - 5 = %d\n",calc(sub, 3, 5));

  return 0;
}

File: perl/demo2.pl

#!/usr/bin/perl
#FUNCTION: 实现判断语句与while循环
$num1 = 1024;
$num2 = 1;
$sum = 0;

if( $num1 != "" && $num2 != "" )
{
  $add = $num1 + $num2;
  print "$num1 + $num2 = $add\n";
}

while($num <= 100)
{
  $sum = $sum + $num;
  $num++;
}

print "1+..+3+...+7+100 = $sum \n";
